sqlmesh/core/model/definition.py

```diff
@@ -1002,9 +1008,11 @@ def depends_on(self) -> t.Set[str]:
 
             query = self.render_query(optimize=False)
             if query is not None:
-                self._depends_on |= d.find_tables(query, dialect=self.dialect)
+                self._depends_on |= d.find_tables(
```

Comment:
I think this is the method in which we should also transform the `depends_on` set by the user.

Reply:
i think it does because it passes in default_catalog and dialect, so this is a normalization entry point

Reply:
If the user defined a depends_on then it should go through the model validator and be normalized. This is handling where it is not set by the user and we get it from the query.